src/Model/ISO2ROI.py

```diff
@@ -0,0 +1,292 @@
+
+        progress_callback.emit(("Validating Datasets", 0))
+        val = ImageLoading.is_dataset_dicom_rt(patient_dict_container.dataset)
+
+        # Stop loading
+        if interrupt_flag.is_set():
+            print("Stopped ISO2ROI")
+            return False
+
+        if not val:
+            # Check if RT struct file is missing. If yes, create one and
+            # add its data to the patient dict container
+            if not patient_dict_container.get("file_rtss"):
+                # Stop loading
+                if interrupt_flag.is_set():
+                    print("Stopped ISO2ROI")
+                    return False
+
+                # Get common directory
+                file_path = patient_dict_container.filepaths.values()
+                file_path = Path(os.path.commonpath(file_path))
+
+                # Get new RT Struct file path
+                file_path = str(file_path.joinpath("rtss.dcm"))
+
+                # Create RT Struct file
+                progress_callback.emit(("Generating RT Structure Set", 30))
+                ds = ROI.create_initial_rtss_from_ct(
+                    patient_dict_container.dataset[0], file_path)
+                ds.save_as(file_path)
+
+                # Add RT Struct file path to patient dict container
+                patient_dict_container.filepaths['rtss'] = file_path
+                filepaths = patient_dict_container.filepaths
+
+                # Add RT Struct dataset to patient dict container
+                patient_dict_container.dataset['rtss'] = ds
+                dataset = patient_dict_container.dataset
+
+                # Set some patient dict container attributes
+                patient_dict_container.set("file_rtss", filepaths['rtss'])
+                patient_dict_container.set("dataset_rtss", dataset['rtss'])
+
+                dicom_tree_rtss = DicomTree(filepaths['rtss'])
+                patient_dict_container.set("dict_dicom_tree_rtss",
+                                           dicom_tree_rtss.dict)
+
+                dict_pixluts = ImageLoading.get_pixluts(
+                    patient_dict_container.dataset)
+                patient_dict_container.set("pixluts", dict_pixluts)
+
+                rois = ImageLoading.get_roi_info(ds)
+                patient_dict_container.set("rois", rois)
+
+                patient_dict_container.set("selected_rois", [])
+                patient_dict_container.set("dict_polygons_axial", {})
+
+        # Get isodose levels to turn into ROIs
+        self.get_iso_levels()
+
+        # Stop loading
+        if interrupt_flag.is_set():
+            print("Stopped ISO2ROI")
+            return False
+
+        # Calculate dose boundaries
+        progress_callback.emit(("Calculating Boundaries", 50))
+        boundaries = self.calculate_boundaries()
+
+        # Return if boundaries could not be calculated
+        if not boundaries:
+            print("Boundaries could not be calculated.")
```

Comment:
Do we need these prints? in line 109 and 114

Reply:
Print statements are included when the ISO2ROI process has to be prematurely stopped for any reason. They were included as an indication to the user of this, as we thought a simple print statement would be a lot cleaner code-wise than including extra pop-up windows, plus print statements do not interrupt the user and what they are doing. If multiple reviewers think that these are not necessary then they can be removed, however we do think that they should stay for reporting purposes.

Reply:
Hm I see... I don't think print could replace a pop-up window if there's an error... Future developers might just think it's a debug print and delete it. Besides, when you packae it into - for example, a .exe program - users cant see the print if running this functionality has error.

Does this action happen in the background (while calculating boundaries user can still interact with other parts of the program)?

Reply:
That's what logging is for.
You write to logs in well known directories (frequently by configuration).
Most trouble shooting in the commercial environment is by reading logs.  And if one can not determine the problem from the logs, one can at least determine where things stopped working.
Logging is your friend.  Logging with levels of severity is your life raft.

In general, the message to the user has to be relatively non technical.  Logs can be considered a note to yourself, although it's best for it to be understandable to fellow programmers, if not support personnel.  I dump stack traces in logs. I say naughty things about other vendors in my logs (when it's evidence that they can't count, for example).

Start using logging.

Reply:
> Hm I see... I don't think print could replace a pop-up window if there's an error... Future developers might just think it's a debug print and delete it. Besides, when you package the program into - for example, a .exe file - users cant see the print if running this functionality has error.

That is true! However as @sjswerdloff says, logging would be better for errors. However adding logging is a big feature in and of itself, and we have other features to focus on that the product owner has requested. Perhaps this could be put to him?

For now the print statements can be deleted, or replaced with dialog boxes, if the majority think that is best.

Reply:
there is now a sqlite file available for logs, is there not?? It is not a user, feature as was pointed out, messages to the screen are for the users, messages to logs are for the technical people. Make life easier for yourself and those who follow. DO not put in logging everywhere, only as the system grows, and put in log level flags from the beginning, with a defined log level in the config file so it is able to be altered.

https://docs.python.org/3/library/logging.html

https://docs.python.org/3/howto/logging.html

google returned several references of using sqlite with the python logger, and logging levels.

Reply:
I did a quick look through the code base and found no mention of logging currently in OnkoDICOM - only in some of the libraries used. I've also been informed since that the print statements were originally as, during development, sometimes the ISO2ROI process would stop with no warning and no indication as to why. Since it all appears to be working now, these can be removed.

Reply:
Looks like these print statements everywhere in the program so i think if we refactor it - better refactor it all in a future work item. If you keep these print. please add a comment in the code, like `# TODO: convert all print to logging`

Reply:
> Looks like these print statements everywhere in the program so i think if we refactor it - better refactor it all in a future work item. If you keep these print. please add a comment in the code, like `# TODO: convert all print to logging`

An excellent way forward.

Reply:
`TODO` comments have been added above print statements.